src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -46,25 +48,35 @@ public function append($id, DomainEventStreamInterface $eventStream)
     {
         $id = (string) $id;
 
-        if (! isset($this->events[$id])) {
-            $this->events[$id] = [];
+        $aggregateEvents = [];
+        if (isset($this->events[$id])) {
+            $aggregateEvents = $this->events[$id];
         }
 
         foreach ($eventStream as $event) {
-            $playhead = $event->getPlayhead();
-            $this->assertPlayhead($this->events[$id], $playhead);
-
-            $this->events[$id][$playhead] = $event;
+            $aggregateEvents = $this->insertEvent($aggregateEvents, $event, $eventStream);
         }
+
+        $this->events[$id] = $aggregateEvents;
```

Comment:
Needed to look a few times before realising you are solving a bug with this :smile: 

I did note however that there is still a minor bug present, but I'll create a separate PR for that.